Pseudocode/Queue/First Non Repeating in Stream/SourceCode.tex

```diff
@@ -0,0 +1,59 @@
+% Set the Page Layout
+\documentclass[12pt]{article}
+\usepackage[inner = 2.0cm, outer = 2.0cm, top = 2.0cm, bottom = 2.0cm]{geometry}
+
+
+% Package to write pseudo-codes
+\usepackage{algorithm}
+
+% Remove the 'end' at the end of the algorithm
+\usepackage[noend]{algpseudocode}
+
+% Define Left Justified Comments
+\algnewcommand{\LeftComment}[1]{\Statex \(\triangleright\) #1}
+
+% Remove the Numbering of the Algorithm
+\usepackage{caption}
+\DeclareCaptionLabelFormat{algnonumber}{Algorithm}
+\captionsetup[algorithm]{labelformat = algnonumber}
+
+\newcommand{\IsNotEmpty}{\textbf{ is not empty, }}
+\newcommand{\Break}{\textbf{break}}
+\newcommand{\Print}{\textbf{Print}}
+
+\begin{document}
+
+    \begin{algorithm}
+    
+      \caption{Find the first non-repeating character when each character of the stream arrives}
+      
+      \begin{algorithmic}[1]
+        \Statex
+        \Function {$First\_Non\_Repeating\_Char$}{$str$}
+            \LeftComment{$frequency$ is a hashmap which stores the frequency of the characters}
+            \LeftComment{$queue$ is a queue of characters}
+            \State{Scan $str$ from left to right and,}
+            \For{each character $char$}
+                \State{$queue$.push($char$)}
+                \State{Increment $frequency$[$char$] by 1}
```

Comment:
Be Precise. A better way is to write it algorithmically, rather than saying it out. Also, I have written a comment (in the gist) to state the assumption that the default frequency of elements which are seen for the first time is zero.